src/WebJobs.Extensions.Http/ClaimsIdentityBindingProvider.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// This provider provides a binding to Type <see cref="ClaimsIdentity"/>.
+    /// </summary>
+    /// <remarks>
+    /// Attempts to bind to an identity in precedence order based on <see cref="AuthorizationLevel"/> levels.
+    /// I.e. if the request is key authenticated (Function/System/Admin) that identity is returned, if the
+    /// request is user authenticated (User) that identity is returned.
+    /// </remarks>
+    internal class ClaimsIdentityBindingProvider : IBindingProvider
```

Comment:
I'm not sold on the utility of this binding. The IEnumerable<ClaimsIdentity> can be gotten straight off of the identity, and us choosing one of them to bind to in the single ClaimsIdentity case just seems like a breaking change waiting to happen.

Reply:
I believe my initial idea in having this binding in addition to the principal binding was to facilitate bindings in other languages (e.g. Node), where the binding is JObject based and the ClaimsPrincipal APIs aren't being used. If we verify the programming model for both .NET and Node can naturally access all identities (in cases where there are multiple) and enumerate them, then I agree we should do the minimum required.